### PR TITLE
Adding the ability to skip compatibility filtering

### PR DESCRIPTION
**Pull request by scbedd**
This feature will be used [on this line](https://github.com/Azure/azure-sdk-for-python/pull/25726/files#r952053598) to prevent 3.11 from _not_ running.

### CODE REVIEW

File: scripts/devops_tasks/setup_execute_tests.py [hunk 1]
```diff
@@ -280,9 +280,9 @@ def execute_global_install_and_test(
         dest="filter_type",
         default='Build',
         help="Filter type to identify eligible packages. for e.g. packages filtered in Build can pass filter type as Build,",
-        choices=['Build', "Docs", "Regression", "Omit_management"]
+        choices=['Build', "Docs", "Regression", "Omit_management", "None"]
     )
-
+
```
Comment (mccoyp) [inline scripts/devops_tasks/setup_execute_tests.py:285]:
```suggestion

```

File: scripts/devops_tasks/setup_execute_tests.py [hunk 2]
```diff
@@ -301,21 +301,23 @@ def execute_global_install_and_test(
     else:
         target_dir = root_dir
 
-    targeted_packages = discover_targeted_packages(args.glob_string, target_dir, "", args.filter_type)
-    compatible_targeted_packages = filter_packages_by_compatibility_override(targeted_packages)
```
Comment (mccoyp) [inline scripts/devops_tasks/setup_execute_tests.py:305]: This seems to be the only use of `filter_packages_by_compatibility_override` -- should we still hold onto that method?

Comment (scbedd) [reply scripts/devops_tasks/setup_execute_tests.py:305]: Check within `discover_targeted_packages` 

https://github.com/Azure/azure-sdk-for-python/blob/fc9e6f937141977b326eaffd5bd168096cee40cd/tools/azure-sdk-tools/ci_tools/functions.py#L112-L145

Comment (mccoyp) [reply scripts/devops_tasks/setup_execute_tests.py:305]: That's using `filter_for_compatibility` though; `filter_packages_by_compatibility_override` seems to be doing something slightly different, but at a glance I'm not sure if it's a duplicate or slightly different in functionality

Comment (scbedd) [reply scripts/devops_tasks/setup_execute_tests.py:305]: I will clean up filter_for_compatibility and filter_packages_by_compatibility_override into a single function. Good catch.

File: tools/azure-sdk-tools/ci_tools/functions.py [hunk 3]
```diff
@@ -42,7 +42,7 @@
     "azure-mgmt-core",
 ]
 
-TEST_COMPATIBILITY_MAP = {"azure-core-tracing-opentelemetry": "<3.10"}
```
Comment (scbedd) [inline tools/azure-sdk-tools/ci_tools/functions.py:45]: Whatever the issue with azure-core-tracing-opentelemetry was, the specifier including py310 is there now, so I'm cleaning it out of this.

File: scripts/devops_tasks/setup_execute_tests.py (hunk 2 above)
Comment (scbedd) [reply scripts/devops_tasks/setup_execute_tests.py:305]: Done! Thanks for the follow-up on this one McCoy!

**Review by mccoyp — APPROVED** [review]

**Pull request merged**